skops/hub_utils/_hf_hub.py

```diff
@@ -482,3 +482,61 @@ def download(
     shutil.copytree(cached_folder, dst)
     if not keep_cache:
         shutil.rmtree(path=cached_folder)
+
+
+def get_model_output(repo_id: str, data: Any, token: Optional[str] = None) -> Any:
+    """Returns the output of the model using Hugging Face Hub's inference API.
+
+    See the :ref:`User Guide <hf_hub_inference>` for more details.
+
+    Parameters
+    ----------
+    repo_id: str
+        The ID of the Hugging Face Hub repository in the form of
+        ``OWNER/REPO_NAME``.
+
+    data: Any
+        The input to be given to the model. This can be a
+        :class:`pandas.DataFrame` or a :class:`numpy.ndarray`. If possible, you
+        should always pass a :class:`pandas.DataFrame` with correct column
+        names.
+
+    token: str, optional
+        The token to be used to call the inference API. Only required if the
+        repository is private.
+
+    Returns
+    -------
+    output: numpy.ndarray
+        The output of the model.
+
+    Notes
+    -----
+    If there are warnings or exceptions during inference, this function raises
+    a :class:`RuntimeError` including the original errors and warnings
+    returned from the server.
+    """
+    model_info = HfApi().model_info(repo_id=repo_id, token=token)
+    if not model_info.pipeline_tag:
+        raise ValueError(
+            f"Repo {repo_id} has no pipeline tag. You should set a valid 'task' in"
+            " config.json and README.md files. This is automatically done for you if"
+            " you pass a valid task to the skops.hub_utils.init() and"
+            " skops.card.metadata_from_util() functions to generate those files."
+        )
+
+    try:
+        inputs = {"data": data.to_dict(orient="list")}
+    except AttributeError:
+        # the input is not a pandas DataFrame
+        inputs = {f"x{i}": data[:, i] for i in range(data.shape[1])}
+        inputs = {"data": inputs}
+
+    res = InferenceApi(repo_id=repo_id, task=model_info.pipeline_tag, token=token)(
+        inputs=inputs
+    )
+
+    if isinstance(res, list):
```

Comment:
Can we rely on the assumption that if the output is a list, everything went well, else it didn't?

Reply:
Yes, on the inference side, we always raise when there's at least one warning or error: https://github.com/huggingface/api-inference-community/blob/5b9970bae8928019dd725f7f16832d9be753df3b/docker_images/sklearn/app/pipelines/tabular_classification.py#L55

Reply:
Okay, and we can be sure that if an error is raised, the API will never return a list. Ideally, I would wish for a response object with a status code but api inference doesn't seem to support that.

On the other hand, when the response _is ok_, we know it's a list because of how the `Pipeline`s are implemented. I guess this introduces some coupling, as future pipelines must always return a list as output too. For now it works, so we can proceed, but it doesn't feel very robust to me.

Reply:
The response comes as a json object, so if there are only values in it, it's a list. But I agree a more complex response or having warnings in the header would be better maybe. For now I'd say we can merge and we can change this if the logic on the backend changes. I don't think we can easily future proof this bit here.